cardillo/rods/_base.py

```diff
@@ -771,20 +774,16 @@ def f_int_el(self, qe, el):
             ############################
             # virtual work contributions
             ############################
+            n = A_IB @ B_n * qwi
             for node in range(self.nnodes_element_r):
                 f_int_el[self.nodalDOF_element_r_u[node]] -= (
-                    self.N_r_xi[el, i, node] * A_IB @ B_n * qwi
+                    self.N_r_xi[el, i, node] * n
```

Comment:
I woul suggest `n_qwi` since this is the internal force vector times the quadrature weight.